src/java.desktop/windows/native/libawt/windows/ShellFolder2.cpp

```diff
@@ -941,16 +977,10 @@ JNIEXPORT jlong JNICALL Java_sun_awt_shell_Win32ShellFolder2_extractIcon
         UINT uFlags = getDefaultIcon ? GIL_DEFAULTICON : GIL_FORSHELL | GIL_ASYNC;
         hres = pIcon->GetIconLocation(uFlags, szBuf, MAX_PATH, &index, &flags);
         if (SUCCEEDED(hres)) {
-            HICON hIconLarge;
-            hres = pIcon->Extract(szBuf, index, &hIconLarge, &hIcon, (16 << 16) + 32);
-            if (SUCCEEDED(hres)) {
-                if (getLargeIcon) {
-                    fn_DestroyIcon((HICON)hIcon);
-                    hIcon = hIconLarge;
-                } else {
-                    fn_DestroyIcon((HICON)hIconLarge);
-                }
+            if (size < 24) {
+                size = 16;
             }
+            hres = pIcon->Extract(szBuf, index, &hIcon, NULL, (16 << 16) + size);
```

Comment:
```suggestion
            hres = pIcon->Extract(szBuf, index, &hIcon, NULL, size);
```
Now you request only one icon.